awx/main/scheduler/task_manager.py

```diff
@@ -519,20 +519,27 @@ def process_pending_tasks(self, pending_tasks):
             if not found_acceptable_queue:
                 logger.debug("{} couldn't be scheduled on graph, waiting for next cycle".format(task.log_format))
 
-    def timeout_approval_node(self):
-        workflow_approvals = WorkflowApproval.objects.filter(status='pending').prefetch_related('workflow_approval_template')
+    def timeout_approval_node(self): # Add websocket stuff for when it transitions to "timed out" (maybe a websocket_emit_status() call)
+        workflow_approvals = WorkflowApproval.objects.filter(status='pending')
         now = tz_now()
         for task in workflow_approvals:
-            # TODO: copy the timeout to the job itself at launch time, not the template
-            approval_timeout_seconds = timedelta(seconds=task.workflow_approval_template.timeout)
-            if task.workflow_approval_template.timeout == 0:
+            # TODO: copy the timeout to the job itself at launch time, not the template <---- Started to implement steps to do this, but unsure...
```

Comment:
This looks exactly right to me, I don't think there's anything else to do here.

If you want to test this, you could:

1.  Make a WF w/ an approval node w/ a 5 minute timeout.
2.  Launch it.
3.  Go edit the `WorkflowApprovalTemplate` and change its timeout from 5 minutes to e.g., 10s
4.  Verify that the timeout on the `WorkflowApproval` still says 5 minutes.

(based on the code here, I'd expect this to work)